### PR TITLE
Rename receipt indexes and remove functions.

### PR DESCRIPTION
**Pull request by arijitAD**

### CODE REVIEW

**Review by i-norden — COMMENTED** [review]
Can we go ahead and just update 00027 and 00028 with these changes instead of creating 00029? We haven't run 00027 yet on our prod database so we can change it.

Looks like this takes care of #12 and #13  but could we also get #14 , and #15 here? You can rebase on top of https://github.com/vulcanize/statediff-migrations/pull/18. It also adds a make target for running the DOWN migrations which we will use for #17

**Comment by i-norden** [comment]
Actually I think I made a mistake in https://github.com/vulcanize/statediff-migrations/pull/18/commits/9bc4d38a4d37b6617e8b0ba182d241b2e431e88a as we have ran 00025 on the prod db, so those changes should in a later/new migration.

File: db/migrations/00029_remove_storage_header_and_state_function.sql [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+DROP FUNCTION "ethHeaderCidByBlockNumber"(bigint);
+
+ALTER INDEX eth.rct_leaf_cid_index
+RENAME TO rct_leaf_mh_index;
+
+ALTER INDEX eth.rct_leaf_mh_key_index
+RENAME TO rct_leaf_cid_index;
+
+ALTER TABLE ONLY eth.receipt_cids
+    RENAME CONSTRAINT receipt_cids_mh_key_fkey TO receipt_cids_leaf_mh_key_fkey;
```
Comment (i-norden) [inline db/migrations/00029_remove_storage_header_and_state_function.sql]: This shouldn't be indented to match style elsewhere

Comment (arijitAD) [reply db/migrations/00029_remove_storage_header_and_state_function.sql]: Done

**Comment by arijitAD** [comment]
> Can we go ahead and just update 00027 and 00028 with these changes instead of creating 00029? We haven't run 00027 yet on our prod database so we can change it.
> 
> Looks like this takes care of #12 and #13 but could we also get #14 , and #15 here? You can rebase on top of #18. It also adds a make target for running the DOWN migrations which we will use for #17

Fixed #14 and #15

File: db/migrations/00025_create_eth_log_cids_table.sql [hunk 2]
```diff
@@ -23,47 +23,36 @@ DROP COLUMN log_contracts,
 ADD COLUMN log_root VARCHAR(66);
 
 CREATE INDEX log_rct_id_index ON eth.log_cids USING btree (receipt_id);
-
```
Comment (i-norden) [inline db/migrations/00025_create_eth_log_cids_table.sql:26]: Changes to this file, unfortunately, need to be made in a new migration as we have already ran 00025 on our prod db.

File: db/migrations/00025_create_eth_log_cids_table.sql [hunk 3]
```diff
@@ -23,47 +23,36 @@ DROP COLUMN log_contracts,
...
+
+ALTER TABLE eth.receipt_cids
+ADD COLUMN topic0s VARCHAR(66)[],
+ADD COLUMN topic1s VARCHAR(66)[],
+ADD COLUMN topic2s VARCHAR(66)[],
+ADD COLUMN topic3s VARCHAR(66)[],
+ADD COLUMN log_contracts VARCHAR(66)[],
+DROP COLUMN log_root;
+
+CREATE INDEX rct_topic0_index ON eth.receipt_cids USING gin (topic0s);
```
Comment (i-norden) [inline db/migrations/00025_create_eth_log_cids_table.sql:44]: I think we need to drop these in the UP section. https://github.com/vulcanize/statediff-migrations/pull/18/files#diff-300ee54917242fcaa78f8d7d0484e902f5cb3925bcaf10c79d41963ab72ed23cR17

```sql
DROP INDEX eth.rct_log_contract_index;
DROP INDEX eth.rct_topic3_index;
DROP INDEX eth.rct_topic2_index;
DROP INDEX eth.rct_topic1_index;
DROP INDEX eth.rct_topic0_index;
```

**Review by i-norden — COMMENTED** [review]
Looks good, just a few nitpicks. And I don't think we can modify 00025 as it was already ran on prod.

File: db/migrations/00025_create_eth_log_cids_table.sql (hunk 3 above)
Comment (arijitAD) [reply db/migrations/00025_create_eth_log_cids_table.sql:44]: @i-norden If we drop the column the index automatically gets deleted. Should I explicitly drop the index before dropping the column?

File: db/migrations/00025_create_eth_log_cids_table.sql (hunk 2 above)
Comment (arijitAD) [reply db/migrations/00025_create_eth_log_cids_table.sql:26]: But I just added the code in `Down` migration. The code of `Up` migration is still the same. Does goose also store the migration in the meta table or just the migration version? I assumed it was just migration verison.

Comment (arijitAD) [reply db/migrations/00025_create_eth_log_cids_table.sql:26]: Screenshot 2021-10-07 at 10 04 20 PM
This is the `goose_db_version` table schema. Doesn't seem to store the migration.

Comment (arijitAD) [reply db/migrations/00025_create_eth_log_cids_table.sql:26]: https://github.com/pressly/goose/blob/0e770f837f7f57ce81942dec9e9e6edb9a068c79/dialect.go#L56

File: db/migrations/00025_create_eth_log_cids_table.sql (hunk 3 above)
Comment (i-norden) [reply db/migrations/00025_create_eth_log_cids_table.sql:44]: No that's an oversight on my part, this is fine. Thanks!

File: db/migrations/00025_create_eth_log_cids_table.sql (hunk 2 above)
Comment (i-norden) [reply db/migrations/00025_create_eth_log_cids_table.sql:26]: Oh I see, that looks good then!

**Review by i-norden — APPROVED** [review]

**Pull request merged**